File: src/main/java/com/vk/tlschema/TLSchemaFileType.java
package com.vk.tlschema;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class TLSchemaFileType extends LanguageFileType {
    public static final TLSchemaFileType INSTANCE = new TLSchemaFileType();

    private TLSchemaFileType() {
        super(TLSchemaLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "TL schema file";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "TL schema file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "tl";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return TLSchemaIcons.FILE;
    }

}

File: src/main/kotlin/com/vk/tlschema/TLSchemaFileType.kt
package com.vk.tlschema

import com.intellij.openapi.fileTypes.LanguageFileType
import javax.swing.Icon

object TLSchemaFileType : LanguageFileType(TLSchemaLanguage.INSTANCE) {
    override fun getName(): String = "TLSchema"

    override fun getDescription(): String = "TL schema file"

    override fun getDefaultExtension(): String = "tl"

    override fun getIcon(): Icon = TLSchemaIcons.FILE
}
